service/raft/raft_group0.cc

```diff
@@ -173,14 +173,17 @@ raft_server_for_group raft_group0::create_server_for_group0(raft::group_id gid,
     auto storage = std::make_unique<raft_sys_table_storage>(_qp, gid, my_addr.id);
     auto& persistence_ref = *storage;
     auto* cl = _qp.proxy().get_db().local().commitlog();
+    auto config = raft::server::configuration {
+        // Dividing by two is to protect against paddings that the
+        // commit log can add for each mutation, as well as
+        // against different commit log limits on different nodes.
+        .max_command_size = cl ? cl->max_record_size() / 2 : 0
```

Comment:
> That would be harder to get correct though than this static check, because the check is only performed when applier_fiber wakes up, which happens only when there are commands to apply. It could happen that add_entry with big command is called after applier_fiber performs the check and decides not to snapshot.

This is exactly the reason why I decided to go with this eager check, but it seem sub-optimal to me too. Otherwise we would either have to somehow wake up `applier_fiber` from `consume_memory()`, or we could extract the code that runs the snapshot into a method and call it in case we are going to block in `consume_memory`. The latter could probably break some invariants.

Reply:
Discussed at the daily meeting, decided to go with the current approach and try to improve it in a separate PR.